### PR TITLE
[G2M] notes - match invalid filename chars and replace with dash

### PR DESCRIPTION
**Pull request by hyx131**

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -75,7 +75,10 @@ module.exports.formatNotes = ({ parsed, version, previous }) => {
 }
 
 module.exports.writeNotes = ({ notes, version, previous, file }) => {
-  const f = file || `release-notes-${previous}-${version}.md`
+  const invalidFNChars = /[\\/:"*?<>|#%$&{}!+`='" ]+/g
+  const p = previous.trim().toLowerCase().replace(invalidFNChars, '-')
+  const v = version.trim().toLowerCase().replace(invalidFNChars, '-')
+  const f = file || `release-notes-${p}-${v}.md`
```
Comment (woozyking) [inline lib/utils.js:81]: why not just apply this whole thing together

```js
const invalidFNChars = /[\\/:"*?<>|#%$&{}!+`='" ]+/g 
const f = (file || `release-notes-${p}-${v}.md`.toLowerCase()).replace(invalidFNChars, '-')
```

Comment (hyx131) [reply lib/utils.js:81]: oh ya that's true lol, just wanna trim `p` and `v` as well

Comment (hyx131) [reply lib/utils.js]: updated

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- `notes` - match invalid filename chars and replace with dash (db25ec9 by hyx131)
```
Comment (woozyking) [inline CHANGELOG.md:9]: have to be careful about the hash when an overriding force push has happened (in this case due to `git commit --amend`)

File: lib/command.notes.js [hunk 3]
```diff
@@ -42,7 +42,7 @@ const handler = ({ base, head, pattern, file, verbose }) => {
     const parsed = parseCommits(logs)
 
     const notes = formatNotes({ parsed, version, previous })
-    writeNotes({ notes, version, previous, file })
+    writeNotes({ notes, version: version.trim(), previous: previous.trim(), file })
```
Comment (woozyking) [inline lib/command.notes.js:45]: this can be done probably at line 37-38 so this gets nicely applied to both `formatNotes` and `writeNotes`

**Review by woozyking — APPROVED** [review]

**Pull request merged**